File: clean_file_paths.php
<?php
/**
 * Script untuk membersihkan path file di desk_bumdes2025.json
 * Menghapus prefix folder uploads/dokumen_bumdes/ dan laporan_keuangan/
 */

function cleanFilePath($path) {
    if (empty($path)) {
        return $path;
    }
    
    // Hapus prefix uploads/dokumen_bumdes/
    if (strpos($path, 'uploads/dokumen_bumdes/') === 0) {
        return str_replace('uploads/dokumen_bumdes/', '', $path);
    }
    
    // Hapus prefix laporan_keuangan/
    if (strpos($path, 'laporan_keuangan/') === 0) {
        return str_replace('laporan_keuangan/', '', $path);
    }
    
    return $path;
}

function cleanBumdesData(&$data) {
    $cleaned = false;
    
    // Jika ada key laporan_keuangan
    if (isset($data['laporan_keuangan']) && is_array($data['laporan_keuangan'])) {
        foreach ($data['laporan_keuangan'] as $key => &$value) {
            if (is_string($value)) {
                $originalValue = $value;
                $value = cleanFilePath($value);
                if ($originalValue !== $value) {
                    $cleaned = true;
                }
            }
        }
    }
    
    // Bersihkan field file lainnya jika ada
    $fileFields = [
        'FileBadanHukum', 'FilePerdes', 'FileSK', 'FileNIB', 'FileLKPP', 
        'FileNPWP', 'FileLaporanKeuangan', 'FileAkta'
    ];
    
    foreach ($fileFields as $field) {
        if (isset($data[$field]) && is_string($data[$field])) {
            $originalValue = $data[$field];
            $data[$field] = cleanFilePath($data[$field]);
            if ($originalValue !== $data[$field]) {
                $cleaned = true;
            }
        }
    }
    
    return $cleaned;
}

function main() {
    $inputFile = 'desk_bumdes2025.json';
    $backupFile = 'desk_bumdes2025_backup.json';
    
    try {
        // Backup file asli
        if (file_exists($inputFile)) {
            echo "Membuat backup: $backupFile\n";
            copy($inputFile, $backupFile);
        }
        
        // Baca file JSON
        echo "Membaca file: $inputFile\n";
        $jsonContent = file_get_contents($inputFile);
        $data = json_decode($jsonContent, true);
        
        if ($data === null) {
            throw new Exception("Error decoding JSON file");
        }
        
        echo "Total data BUMDes: " . count($data) . "\n";
        
        // Bersihkan data
        $cleanedCount = 0;
        foreach ($data as $i => &$bumdes) {
            if (cleanBumdesData($bumdes)) {
                $cleanedCount++;
                $namabumdesa = isset($bumdes['namabumdesa']) ? $bumdes['namabumdesa'] : 'Unknown';
                echo "Membersihkan data BUMDes #" . ($i+1) . ": $namabumdesa\n";
            }
        }
        
        // Simpan file yang sudah dibersihkan
        echo "Menyimpan file yang sudah dibersihkan: $inputFile\n";
        $cleanedJson = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE);
        file_put_contents($inputFile, $cleanedJson);
        
        echo "\n✅ Selesai!\n";
        echo "📊 Total data yang dibersihkan: $cleanedCount\n";
        echo "💾 File backup: $backupFile\n";
        echo "✨ File bersih: $inputFile\n";
        
    } catch (Exception $e) {
        echo "❌ Error: " . $e->getMessage() . "\n";
    }
}

main();
?>

File: clean_file_paths.py
#!/usr/bin/env python3
"""
Script untuk membersihkan path file di desk_bumdes2025.json
Menghapus prefix folder uploads/dokumen_bumdes/ dan laporan_keuangan/
"""

import json
import re
import os

def clean_file_path(path):
    """
    Membersihkan path file dengan menghapus prefix folder
    """
    if not path:
        return path
    
    # Hapus prefix uploads/dokumen_bumdes/
    if path.startswith('uploads/dokumen_bumdes/'):
        return path.replace('uploads/dokumen_bumdes/', '')
    
    # Hapus prefix laporan_keuangan/
    if path.startswith('laporan_keuangan/'):
        return path.replace('laporan_keuangan/', '')
    
    return path

def clean_bumdes_data(data):
    """
    Membersihkan data BUMDes dengan menghapus prefix folder dari path file
    """
    if isinstance(data, dict):
        # Jika ada key laporan_keuangan
        if 'laporan_keuangan' in data:
            laporan_keuangan = data['laporan_keuangan']
            if isinstance(laporan_keuangan, dict):
                for key, value in laporan_keuangan.items():
                    if isinstance(value, str):
                        laporan_keuangan[key] = clean_file_path(value)
        
        # Bersihkan field file lainnya jika ada
        file_fields = [
            'FileBadanHukum', 'FilePerdes', 'FileSK', 'FileNIB', 'FileLKPP', 
            'FileNPWP', 'FileLaporanKeuangan', 'FileAkta'
        ]
        
        for field in file_fields:
            if field in data and isinstance(data[field], str):
                data[field] = clean_file_path(data[field])
    
    return data

def main():
    """
    Fungsi utama untuk membersihkan file JSON
    """
    input_file = 'desk_bumdes2025.json'
    backup_file = 'desk_bumdes2025_backup.json'
    
    try:
        # Backup file asli
        if os.path.exists(input_file):
            print(f"Membuat backup: {backup_file}")
            with open(input_file, 'r', encoding='utf-8') as f:
                original_data = f.read()
            with open(backup_file, 'w', encoding='utf-8') as f:
                f.write(original_data)
        
        # Baca file JSON
        print(f"Membaca file: {input_file}")
        with open(input_file, 'r', encoding='utf-8') as f:
            data = json.load(f)
        
        print(f"Total data BUMDes: {len(data)}")
        
        # Bersihkan data
        cleaned_count = 0
        for i, bumdes in enumerate(data):
            original_bumdes = json.dumps(bumdes, sort_keys=True)
            cleaned_bumdes = clean_bumdes_data(bumdes)
            if json.dumps(cleaned_bumdes, sort_keys=True) != original_bumdes:
                cleaned_count += 1
                print(f"Membersihkan data BUMDes #{i+1}: {bumdes.get('namabumdesa', 'Unknown')}")
        
        # Simpan file yang sudah dibersihkan
        print(f"Menyimpan file yang sudah dibersihkan: {input_file}")
        with open(input_file, 'w', encoding='utf-8') as f:
            json.dump(data, f, indent=4, ensure_ascii=False)
        
        print(f"\n✅ Selesai!")
        print(f"📊 Total data yang dibersihkan: {cleaned_count}")
        print(f"💾 File backup: {backup_file}")
        print(f"✨ File bersih: {input_file}")
        
    except FileNotFoundError:
        print(f"❌ Error: File {input_file} tidak ditemukan!")
    except json.JSONDecodeError as e:
        print(f"❌ Error: File JSON tidak valid - {e}")
    except Exception as e:
        print(f"❌ Error tidak terduga: {e}")

if __name__ == "__main__":
    main()
